### PR TITLE
Added C++ implementation to the first 3 sum_of_squares files

### PR DESCRIPTION
**Pull request by BradeyLounsbury**
I tried to keep the output as close to the original files as possible but I think the time functions available with C++ aren't as precise as the time functions in python so the data output from the C++ files are only out to 6 decimal places.

Also, the C++ implementations seemed to scale linearly whereas the python implementation were exponential.  With lower max values, python was faster but as the max increased, C++ began to close the gap until it eventually surpassed python.

### CODE REVIEW

File: sum_of_squares/SumOfPerfectSquares-dp-static.cc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+int sum_of_squares(int n) {
+    int existing_count = g_count.size();
+    if (n < existing_count)
+    {
+        return g_count[n];
+    }
+    for (size_t i = existing_count; i < n+1; i++)
+    {
+        g_count.push_back(INT_MAX);
```
Comment (krerkkiat) [inline sum_of_squares/SumOfPerfectSquares-dp-static.cc:22]: This file cannot be compiled on the school linux server. Could you please double check?

File: sum_of_squares/SumOfPerfectSquares-dp.cc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#include <vector>
+#include <string>
+
+using namespace std;
+
+int sum_of_squares(int n) {
+    vector<int> g_count;
+    for (size_t i = 0; i < n+1; i++)
+    {
+        g_count.push_back(INT_MAX);
```
Comment (krerkkiat) [inline sum_of_squares/SumOfPerfectSquares-dp.cc:16]: This file cannot be compiled on the school linux server. Could you please double check?

File: sum_of_squares/SumOfPerfectSquares-math.cc [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#include <iostream>
```
Comment (krerkkiat) [inline sum_of_squares/SumOfPerfectSquares-math.cc:1]: Note that there is already a translation for this version which is introduced by #145

**Review by krerkkiat — CHANGES_REQUESTED** [review]
Please make sure that the source code can be compiled on school linux server. Note that `SumOfPerfectSquares-math.cc` is introduced in #145 already,.

**Comment by krerkkiat** [comment]
another dynamic programming version introduced in #169

**Comment by BradeyLounsbury** [comment]
I got an compilation error saying INT_MAX wasn't defined so I changed that and it should compile now

**Pull request merged**